features/step_definitions/branch_steps.rb

```diff
@@ -80,7 +80,16 @@
 end
 
 
+Given(/^I am on the "(.+?)" branch with "(.+?)" as the previous Git branch/) do |current_branch, previous_branch|
+  step %(I am on the "#{previous_branch}" branch)
+  step %(I am on the "#{current_branch}" branch)
+end
+
 
+Given(/^my previous Git branch is now "(.+?)"/) do |previous_branch|
+  run 'git checkout -'
+  expect(current_branch_name).to eql previous_branch
```

Comment:
This command leaves the workspace in a different state than it found it. As long as we stick to a particular order of commands that issue will be hidden, but I think its better to do another `git checkout -` at the end to restore the current branch here.